python/lsst/pipe/base/executionButlerBuilder.py

```diff
@@ -126,26 +124,30 @@ def _export(butler: Butler, collections: Optional[Iterable[str]], exports: Set[D
     return yamlBuffer
 
 
-def _setupNewButler(butler: Butler, outputLocation: str, dirExists: bool) -> Butler:
+def _setupNewButler(butler: Butler, outputLocation: ButlerURI, dirExists: bool) -> Butler:
     # Set up the new butler object at the specified location
     if dirExists:
-        if os.path.isfile(outputLocation):
-            os.remove(outputLocation)
-        else:
-            shutil.rmtree(outputLocation)
-    os.mkdir(outputLocation)
+        # Remove the existing table, if the code got this far and this exists
+        # clobber must be true
+        executionRegistry = outputLocation.join("gen3.sqlite3")
+        if executionRegistry.exists():
+            executionRegistry.remove()
+    else:
+        outputLocation.mkdir()
 
     # Copy the existing butler config, modifying the location of the
     # registry to the specified location.
     # Preserve the root path from the existing butler so things like
     # file data stores continue to look at the old location.
-    config = ButlerConfig(butler._config)
-    config["registry", "db"] = f"sqlite:///{outputLocation}/gen3.sqlite3"
-    config["root"] = butler._config.configDir.ospath
+    config = Config(butler._config)
+    config["root"] = outputLocation.geturl()
+    config["registry", "db"] = "sqlite:///<butlerRoot>/gen3.sqlite3"
+    # record the current root of the datastore if it is specified relative
+    # to the butler root
+    if config.get(("datastore", "root")) == BUTLER_ROOT_TAG:
+        config['datastore', 'root'] = butler._config.configDir.geturl()
```

Comment:
Just remembered. I think we want this execution butler to also enable 
```python
config["datastore", "trust_get_request"] = True
```
otherwise the distributed workflow won't function.

Reply:
PS the string quoting is inconsistent on this line relative to the others.